Issue/12750 blaze survey

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12750 
<!-- Id number of the GitHub issue this PR addresses. -->

⚠️ Requires [these FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3105) changes to work

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Adds a new feedback request UI component in Blaze campaign creation success bottom sheet. Upon interacting with either 👍🏼 or 👎🏼 from the feedback component, a survey webview will be displayed

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

### Testing information
Apply this patch to disable the Objective feature flag and be able to create a campaign without setting a mandatory objective value:

```diff
Subject: [PATCH] Disable Objective feature flag in debug
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt	(revision c24bfbad0ba216ce633a385bb80b89cf3712bf2d)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt	(date 1727972939935)
@@ -25,11 +25,11 @@
                 PackageUtils.isDebugBuild() || context != null && PackageUtils.isBetaBuild(context)
             }
 
+            OBJECTIVE_SECTION -> false
             WC_SHIPPING_BANNER,
             BETTER_CUSTOMER_SEARCH_M2,
             ORDER_CREATION_AUTO_TAX_RATE,
-            REVAMP_WOO_SHIPPING,
-            OBJECTIVE_SECTION -> PackageUtils.isDebugBuild()
+            REVAMP_WOO_SHIPPING -> PackageUtils.isDebugBuild()
 
             NEW_SHIPPING_SUPPORT,
             INBOX,

```

1. Create a Blaze campaign
2. In the success bottom sheet click on any of the 👍🏼 or 👎🏼
3. Check the survey is loaded 
4. Check in the Android logcat the following log: 
`🔵 Tracked: blaze_campaign_creation_feedback, Properties: {"satisfied":true...`

### The tests that have been performed

The above ☝🏼. Created a Blaze campaign and opened the survey form

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/user-attachments/assets/c78eec3f-bb48-4e6a-9009-5ecb7004c4df

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Adds new feedback module at the bottom of Blaze success sheet

JorgeMucientes committed:
Track tapping on positive or negative feedback

JorgeMucientes committed:
Add logic to only show feedback request if user has at least 2 campaigns

JorgeMucientes committed:
Extract RequestFeedback composable to a reusable component

JorgeMucientes committed:
Use new generic compose component for feedback in AI product creation

JorgeMucientes committed:
Fix detekt indentation issue

dangermattic commented:
<!--
  0 Errors
  2 Warnings: Class <code>ShouldShowFeedback...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ShouldShowFeedbackRequest</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/239">20.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Send objective null to avoid making the API fail

While objective feature is WIP we were sending objective = "" in the create campaign request. Empty values are not allowed for this field. So until the feature is finished we must avoid sending the objective value at all in the create campaign request.

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes committed:
Fix tracking property

JorgeMucientes committed:
Updates release notes

JorgeMucientes committed:
Revert accidental change

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/BlazeRepository.kt:
```diff
@@ -322,7 +322,7 @@ class BlazeRepository @Inject constructor(
                 },
                 isEndlessCampaign = campaignDetails.budget.isEndlessCampaign,
-                objectiveId = campaignDetails.objectiveId
+                objectiveId = if (FeatureFlag.OBJECTIVE_SECTION.isEnabled()) campaignDetails.objectiveId else null
```
This is unrelated to the Blaze survey changes. But we need to set objective to null to ensure users can create campaigns successfully while the objective section is WIP. Otherwise we'll be sending `objective = ""` in the campaign creation API request and this will make the request fail. 

This change goes alongside [this change from FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3105) that need to be merged first.

irfano replied:
Good! The campaign flow worked successfully when the feature flag was disabled.

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/success/BlazeCampaignSuccessBottomSheetFragment.kt:
```diff
@@ -4,19 +4,28 @@ import android.os.Bundle
     }
+
+    private fun onFeedbackRequestTapped(isPositive: Boolean) {
+        ChromeCustomTabUtils.launchUrl(requireContext(), BLAZE_CAMPAIGN_CREATION_SURVEY_URL_I1)
```
I think we should also dismiss the bottom sheet after opening the webview. Otherwise, it remains visible after completing the survey.

irfano reviewed (approved):
LGTM! 👍🏻 I added only a minor suggestion.

JorgeMucientes committed:
Update fluxC changeset

JorgeMucientes committed:
Dismiss bottom sheet when url is loaded

JorgeMucientes commented:
Thanks for the review @irfano. I've applied your suggestion 👍🏼
